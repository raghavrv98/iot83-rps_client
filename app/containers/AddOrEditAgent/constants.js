/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
