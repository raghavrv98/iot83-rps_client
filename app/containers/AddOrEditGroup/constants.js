/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
