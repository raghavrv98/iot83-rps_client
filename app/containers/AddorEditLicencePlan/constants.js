/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";