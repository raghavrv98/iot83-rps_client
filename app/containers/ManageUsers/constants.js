/*
 *
 * ManageUsers constants
 *
 */

export const DEFAULT_ACTION = "app/ManageUsers/DEFAULT_ACTION";

export const GET_LIST = "app/ManageUsers/GET_LIST";
export const GOT_LIST = "app/ManageUsers/GOT_LIST";
export const NOT_GOT_LIST = "app/ManageUsers/NOT_GOT_LIST";


export const DELETE_REQUEST = "app/ManageUsers/DELETE_REQUEST";
export const DELETE_REQUEST_SUCCESS = "app/ManageUsers/DELETE_REQUEST_SUCCESS";
export const DELETE_REQUEST_FAILURE = "app/ManageUsers/DELETE_REQUEST_FAILURE";