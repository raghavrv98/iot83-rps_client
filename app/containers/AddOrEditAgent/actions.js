/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
