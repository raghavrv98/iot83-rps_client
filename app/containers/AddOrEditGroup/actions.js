/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
