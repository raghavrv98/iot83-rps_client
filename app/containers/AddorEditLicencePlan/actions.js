/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}