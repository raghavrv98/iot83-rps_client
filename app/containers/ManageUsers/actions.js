/*
 *
 * ManageUsers actions
 *
 */

import { DEFAULT_ACTION ,GET_LIST,DELETE_REQUEST} from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function getUserList(payload) {
  return {
    type: GET_LIST,
    payload
  };
}


export function deleteHandler(id) {
  return {
    type: DELETE_REQUEST,
    id
  }
}

