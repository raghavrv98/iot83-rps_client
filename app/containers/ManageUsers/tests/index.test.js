import React from 'react';
import { shallow } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

import { ManageUsers, mapDispatchToProps } from '../index';
import { getUserList, deleteHandler } from '../actions';
import { GET_LIST, DELETE_REQUEST } from "../constants";
import history from 'utils/history';
const mockGroupList = {
  "totalCount": 1,
  "objects": [
    {
      "id": "bb157d6738bb4886b77fd25635301010",
      "firstName": "Skidanenko",
      "lastName": "Constantine",
      "email": "admin@pmmp.com",
      "company": "PMMP.",
      "userRole": [
        {
          "id": "c2278f01c66b4ebcb67c8bad7f273c6c",
          "role": {
            "id": "e9ef98a8afaf4b06b1942159b8418e3a",
            "name": "SYSTEM_ADMIN",
            "type": "SYSTEM_ROLE"
          }
        }
      ],
      "compartment": [
        {
          "id": "9f1c2f19a97d4a01bebec08ddd206fe2",
          "name": "DEFAULT",
          "description": "Default container"
        }
      ],
      "name": "Skidanenko Constantine",
      "internalUser": true,
      "enabled": true
    }
  ]
}

describe('<ManageUsers />', () => {

  it('should test component when render', () => {
    spyOn(ManageUsers.prototype, 'render').and.callThrough();
    const wrapper = shallow(<ManageUsers getUserList={getUserList} list={mockGroupList} />);
    expect(wrapper).toBeDefined();
    expect(ManageUsers.prototype.render).toHaveBeenCalledTimes(2);
  })

  it('should test component when recieve props', () => {
    spyOn(ManageUsers.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<ManageUsers getUserList={getUserList} list={undefined} listError={undefined} deleteSuccess={undefined} deleteFailure={undefined} />);
    expect(wrapper).toBeDefined();
    wrapper.setState({ userList: mockGroupList });
    wrapper.setProps({ list: mockGroupList, listError: "No Error Found.", deleteSuccess: "bb157d6738bb4886b77fd25635301010", deleteFailure: "No Error Found in delete." });
    expect(ManageUsers.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  })

  // it("clicking add button trigger onClick event", async () => {
  //   let expectedCount = history.length + 1
  //   const wrapper = shallow(<ManageUsers getUserList={getUserList} history={history} />);
  //   let addButton = wrapper.find("button.btn.btn-create");
  //   expect(addButton.length).toBe(1);
  //   addButton.simulate('click');
  //   expect(history.length).toEqual(expectedCount);
  //   expect(history.location.pathname).toEqual("/addOrEditUser");
  // });

  it('match snapshots', () => {
    const wrap = shallow(<ManageUsers deleteHandler={deleteHandler} getUserList={getUserList} />)
    expect(wrap).toMatchSnapshot()
  })

  it("Function testing on mapDispatchToProps", () => {
    const mockParams = {
      params: {
        id: "demo",
        payload: "demo"
      }
    };
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getUserList(
      mockParams.params.payload
    );
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: GET_LIST,
      payload: "demo"
    });
    mapDispatchToProps(dispatch).deleteHandler(
      mockParams.params.id
    );
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: DELETE_REQUEST,
      id: "demo"
    });
  });
});
