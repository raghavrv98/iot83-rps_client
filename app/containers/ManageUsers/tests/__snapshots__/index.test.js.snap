// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageUsers /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManageUsers
    </title>
    <meta
      content="Description of ManageUsers"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p>
        <FormattedMessage
          defaultMessage="Identity Access Management"
          id="app.common.messages.iam"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="All Available Users"
          id="app.containers.ManageUsers.title"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonReactTable"
  />
</div>
`;
